### DISCUSSION TITLE
Difference between types `TEXT` and `VARCHAR`?

### DISCUSSION THREAD

**Discussion by b2m9** [supabase/supabase#8107] [Questions] [answered]
What's the difference between the data types `TEXT` and `VARCHAR` in Supabase? I always thought that both are virtually the same in Postgres, but Supabase decides to expose both types in the dashboard interface (but not types `CHAR(n)` or `VARCHAR(n)`). Why is that? What is the actual difference between the two types (size, performance, ...)?

Disclaimer: I'm not an expert in SQL or Postgres.

**Comment by GaryAustin1** [comment, marked as answer]
No difference.
https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-char-varchar-text/
Don’t know why they did not implement varchar(n) other than it would involve a new field to get the number and format it.

**Reply by b2m9** [reply]
Thanks, noted. So it's just an odd choice to show `TEXT` and `VARCHAR` but not `VARCHAR(n)` or `CHAR(n)`. Good, I assumed I'm missing some deeper reason behind all this.
